kube-client/Cargo.toml

```diff
@@ -47,7 +47,8 @@ http = "0.2.5"
 http-body = { version = "0.4.2", optional = true }
 either = { version = "1.6.1", optional = true }
 thiserror = "1.0.29"
-futures = { version = "0.3.17", optional = true }
+futures-channel = { version = "0.3.17", features = ["sink"], optional = true }
+futures-util = { version = "0.3.17", features = ["sink"], optional = true }
```

Comment:
while i like the trimming of dependencies we don't use, the use of all their peripheral crates (when futures have a facade-crate model) makes our documentation look less crisp (can no longer just point to futures, docs need to point at one or more sub-crates) and it feels slightly wrong to push that hierarchy complexity onto us + users.

would it make sense to use `futures` directly without [default features](https://github.com/rust-lang/futures-rs/blob/2e30ec3f5d4a3de2454a709acf44667fc10660bf/futures/Cargo.toml#L35-L43) to achieve a similar result?

Reply:
> makes our documentation look less crisp

Another wrinkle that I was not aware of was that rustdoc does not include development dependencies when generating the links, so even having `futures` as a development dependency is not sufficient to keep the documentation in its current form.

> would it make sense to use `futures` directly without default features to achieve a similar result?

I think it would be fine, but I do not have a strong opinion either way. My PR was largely due to the above misconception about rustdoc behavior, and I completely agree with your feeling that it seems wrong to use the `futures` subcrates in official documentation.

Realistically, saving 4 dependencies by using the `futures` subcrates is not all that significant, so I will close this PR for the time being unless there are any that would really like to have this included. Nonetheless, thank you for your time in reviewing the changes.

Reply:
i do appreciate you taking the time to do this. was not aware that the ecosystem was so consistent in the futures ecosystem use (most tend to only use util + core, and occasionally channel) so i think there is real value in aligning.

I think it's actually possible to get around the documentation issue by just referring to `futures` even though the file we are documenting does not have it in scope. we can use the same trick we use internally by referring to a crate in dev-dependencies (and have full futures listed in dev-dependencies)

Reply:
> I think it's actually possible to get around the documentation issue by just referring to `futures`

Interesting, I did not know that that was a possibility. I will give it a try and reopen the PR if that is alright with you.